Add density to Darcy source in NSCLS (#2005)
chaos-polymtl/lethe: doc/source/parameters/cfd/velocity_source.rst

  end

* The ``Darcy type`` parameter specifies the type of Darcy penalization term to be applied to the Navier-Stokes equations. The options are ``none`` or ``phase_change``. The ``phase_change`` model uses the values of the ``Darcy penalty liquid``  and ``Darcy penalty solid`` set-up within the ``phase change`` subsection of the :doc:`physical_properties`.

  .. caution::
    The phase change Darcy model does not currently have a Cahn-Hilliard implementation.

* The ``enable Darcy multiply by density`` parameter enables the multiplication by the density within the Darcy force term (:math:`\boldsymbol{F}_\mathrm{Darcy}`). This is for dimensional consistency when solving the pressure (:math:`p`) rather than the kinematic pressure (:math:`p^* = p / \rho`, with :math:`\rho` the density of the fluid) in the momentum equation. This parameter should be used when coupling of the :doc:`CLS equation<../../theory/multiphase/cfd/cls>` with the :doc:`incompressible Navier-Stokes equations<../../theory/multiphysics/fluid_dynamics/navier-stokes>`.

  .. math::
    \boldsymbol{F}_\mathrm{Darcy} = \left(\rho K\right)_\mathrm{eff} \boldsymbol{u}

  where

>   * :math:`\left(\rho K\right)_\mathrm{eff} = (1-\phi)\rho_0 [\alpha_\mathrm{l}K_\mathrm{0,l} + (1-\alpha_\mathrm{l})K_\mathrm{0,s}] + \phi \rho_1 [\alpha_\mathrm{l}K_\mathrm{1,l} + (1-\alpha_\mathrm{l})K_\mathrm{1,s}]` is the effective Darcy penalization coefficient, obtained by weighting the product of the Darcy penalty :math:`\left(K_\mathrm{i} [=] T^{-1}\right)` and the density :math:`\left(\rho_\mathrm{i}[=] M L^{-3}\right)` by the phase indicator (:math:`\phi`) and the liquid fraction (:math:`\alpha_\mathrm{l}`), and;

Reviewer: ```suggestion
  * :math:`\left(\rho K\right)_\mathrm{eff} = (1-\phi)\rho_0 K_\mathrm{0} + \phi \rho_1 K_\mathrm{1}` is the effective Darcy penalization coefficient. It corresponds to the weighted product of the Darcy penalty :math:`\left(K_\mathrm{i} [=] T^{-1}\right)` and the density :math:`\left(\rho_\mathrm{i}[=] M L^{-3}\right)` by the phase indicator (:math:`\phi`). If the ``phase_change`` model is enabled in either fluid, the Darcy penalty in this fluid is computed with the values in the liquid and solid, and by the liquid fraction :math:`\alpha_\mathrm{l}`: :math:`K_\mathrm{i} = \alpha_\mathrm{l}K_\mathrm{i,l} + (1-\alpha_\mathrm{l})K_\mathrm{i,s}`;
```

.. code-block:: text

  subsection velocity source
    set Darcy type                       = none
    set enable Darcy multiply by density = false
  end

* The ``Darcy type`` parameter specifies the type of Darcy penalization term to be applied to the Navier-Stokes equations. The options are ``none`` or ``phase_change``. The ``phase_change`` model uses the values of the ``Darcy penalty liquid``  and ``Darcy penalty solid`` set-up within the ``phase change`` subsection of the :doc:`physical_properties`.

  .. caution::
    The phase change Darcy model does not currently have a Cahn-Hilliard implementation.

> * The ``enable Darcy multiply by density`` parameter enables the multiplication by the density within the Darcy force term (:math:`\vec{F}_\mathrm{Darcy}`). This is for dimensional consistency when solving the pressure (:math:`p`) rather than the kinematic pressure (:math:`p^* = p / \rho`, with :math:`\rho` the density of the fluid) in the momentum balance. This is the case for the coupling of the :doc:`CLS equation<../../theory/multiphase/cfd/cls>` with the :doc:`incompressible Navier-Stokes equations<../../theory/multiphysics/fluid_dynamics/navier-stokes>`.

Reviewer 2: ```suggestion
* The ``enable Darcy multiply by density`` parameter enables the multiplication by the density within the Darcy force term (:math:`\vec{F}_\mathrm{Darcy}`). This is for dimensional consistency when solving the pressure (:math:`p`) rather than the kinematic pressure (:math:`p^* = p / \rho`, with :math:`\rho` the density of the fluid) in the momentum equation. This parameter should be used when coupling of the :doc:`CLS equation<../../theory/multiphase/cfd/cls>` with the :doc:`incompressible Navier-Stokes equations<../../theory/multiphysics/fluid_dynamics/navier-stokes>`.
```

  .. math::
    \vec{F}_\mathrm{Darcy} = \tilde{\rho} \tilde{K} \vec{u}

  where

  * :math:`\tilde{\rho} = (1-\phi)\rho_0 + \phi \rho_1` is the phase indicator (:math:`\phi`) weighted density;
  * :math:`\tilde{K} = (1-\phi)[\alpha_\mathrm{l}K_\mathrm{0,l} + (1-\alpha_\mathrm{l})K_\mathrm{0,s}] + \phi[\alpha_\mathrm{l}K_\mathrm{1,l} + (1-\alpha_\mathrm{l})K_\mathrm{1,s}]` is the Darcy penalty weighted by the liquid fraction (:math:`\alpha_\mathrm{l}`) and phase indicator, and;

>   * :math:`\vec{u}` is the velocity vector.

Reviewer: I wonder if we should do this or treat rho*K as one property: 

$\hat{\rho K} =  \phi (\rho K)_1 - (1-\phi)( \rho K)_0$

(like Justin mentioned in issue https://github.com/chaos-polymtl/lethe/issues/1903)

Reviewer: Also, can you add the dimensions of K?

Reviewer 2: Yeah i don't know. That's a good question. Do we weight (rho*K), or do we weight rho and weight K... I don't have a good answer to this.

Author: I'll try both with the test and see how it influences the results and convergence :)

Author: Having a linear interpolation slightly improves the convergence for the test case so I made the change. :)
